colour/volume/macadam_limits.py

```diff
@@ -142,3 +143,197 @@ def is_within_macadam_limits(
     simplex = np.where(simplex >= 0, True, False)
 
     return simplex
+
+
+def macadam_limits(target_brightness, illuminant=()):
+    """
+    Wavelength reaches from 360 to 830 nm, in within the program it is
+    handled as 0 to 470. Beyond the references this program is very fast,
+    because the possible optimums are not simply tested step by step but
+    more effectively targeted by steps of power of two. The wavelengths
+    left and right of a rough optimum are fitted by a rule of proportion,
+    so that the wished brightness will be reached exactly.
+
+    Parameters
+    ----------
+    target_brightness : floating point
+        brightness has to be between 0 and 1
+
+    illuminant: object
+        illuminant must be out of colorimetry.MSDS_CMFS['XXX']
+        If there is no illuminant or it has the wrong form,
+        the illuminant SDS_ILLUMINANTS['E']
```

Comment:
We are expecting the examples to be run and serve as doctests, those cannot be run. I would make sure that they actually do. The plotting code might be better in a dedicated module in `colour.examples.volume`.